pytrickle/frames.py

```diff
@@ -179,4 +244,235 @@ def with_monotonic_timestamps(cls, frames: List[AudioFrame], request_id: str, st
+        )
+
+    @staticmethod
+    def create_processed_frame(processed_tensor: torch.Tensor, original_frame: VideoFrame) -> VideoFrame:
+        """Create a processed VideoFrame preserving timing from original."""
+        return VideoFrame(
+            tensor=processed_tensor,
+            timestamp=original_frame.timestamp,
+            time_base=original_frame.time_base
+        )
+
+    @staticmethod
+    def convert_trickle_audio_to_av(frame: AudioFrame) -> av.AudioFrame:
+        """Convert pytrickle AudioFrame to av.AudioFrame."""
+        try:
+            samples = frame.samples
+            
+            # Handle different audio format requirements
+            if frame.format.endswith('p'):
+                # Planar format - channels are separated (channels, samples)
+                if samples.ndim == 1:
+                    samples = samples.reshape(1, -1)
+            else:
+                # Packed format - channels are interleaved
+                if samples.ndim == 2 and samples.shape[0] > 1:
+                    # Convert (channels, samples) to (samples, channels) for packed format
+                    samples = samples.T
+                elif samples.ndim == 1:
+                    # Keep 1D for mono packed format or reshape for multi-channel
+                    if frame.layout != 'mono':
+                        # For non-mono, interpret as interleaved samples
+                        pass  # Keep as-is for now
+            
+            av_frame = av.AudioFrame.from_ndarray(samples, format=frame.format, layout=frame.layout)
+            av_frame.sample_rate = frame.rate
+            av_frame.pts = frame.timestamp
+            av_frame.time_base = frame.time_base
+            return av_frame
+            
+        except Exception as e:
+            # If conversion fails, create a simple dummy frame for now
+            logger.warning(f"Audio conversion failed ({e}), creating dummy frame")
+            dummy_samples = np.zeros((1, 1024), dtype=np.int16)
+            av_frame = av.AudioFrame.from_ndarray(dummy_samples, format='s16', layout='mono')
+            av_frame.sample_rate = frame.rate
+            av_frame.pts = frame.timestamp
+            av_frame.time_base = frame.time_base
+            return av_frame
+
+    @staticmethod
+    def convert_av_audio_to_trickle(av_frame: av.AudioFrame, original_frame: AudioFrame) -> AudioFrame:
+        """Convert av.AudioFrame back to pytrickle AudioFrame with original timing."""
+        return AudioFrame.from_av_audio(av_frame)
+
+
+class FrameConversionMixin:
+    """Mixin class that provides frame conversion methods for stream processors."""
+    
+    def convert_trickle_to_av(self, frame: VideoFrame) -> av.VideoFrame:
+        """Convert pytrickle VideoFrame to av.VideoFrame."""
+        return FrameProcessor.convert_trickle_to_av(frame)
+    
+    def convert_av_to_trickle(self, av_frame: av.VideoFrame, original_frame: VideoFrame) -> VideoFrame:
```

Comment:
same as above
```suggestion
    def convert_av_video_to_trickle(self, av_frame: av.VideoFrame, original_frame: VideoFrame) -> VideoFrame:
```

Reply:
Fixed in https://github.com/livepeer/pytrickle/commit/70a0ede3859a13661aa4f86560500e13dc8a4f89 (see above)